### PR TITLE
Identity environmnet credential cert password

### PR DESCRIPTION
**Pull request by xiangyan99**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by xiangyan99** [comment]
#24652

File: sdk/identity/azure-identity/azure/identity/aio/_credentials/environment.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def __init__(self, **kwargs: "Any") -> None:
                 client_id=os.environ[EnvironmentVariables.AZURE_CLIENT_ID],
                 tenant_id=os.environ[EnvironmentVariables.AZURE_TENANT_ID],
                 certificate_path=os.environ[EnvironmentVariables.AZURE_CLIENT_CERTIFICATE_PATH],
+                password=os.environ.get(EnvironmentVariables.AZURE_CLIENT_CERTIFICATE_PASSWORD, None),
```
Comment (chlowell) [inline sdk/identity/azure-identity/azure/identity/aio/_credentials/environment.py:63]: `dict.get()` returns `None` by default
```suggestion
                password=os.environ.get(EnvironmentVariables.AZURE_CLIENT_CERTIFICATE_PASSWORD),
```

**Review by chlowell — APPROVED** [review]

File: sdk/identity/azure-identity/azure/identity/_credentials/environment.py [hunk 2]
```diff
@@ -79,6 +79,7 @@ def __init__(self, **kwargs):
                 client_id=os.environ[EnvironmentVariables.AZURE_CLIENT_ID],
                 tenant_id=os.environ[EnvironmentVariables.AZURE_TENANT_ID],
                 certificate_path=os.environ[EnvironmentVariables.AZURE_CLIENT_CERTIFICATE_PATH],
+                password=os.environ.get(EnvironmentVariables.AZURE_CLIENT_CERTIFICATE_PASSWORD, None),
```
Comment (mccoyp) [inline sdk/identity/azure-identity/azure/identity/_credentials/environment.py:82]: To align with async code:
```suggestion
                password=os.environ.get(EnvironmentVariables.AZURE_CLIENT_CERTIFICATE_PASSWORD),
```

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 ### Features Added
 
+- Environment credential added `AZURE_CLIENT_CERTIFICATE_PASSWORD` supports for the cert password    ([#24652](https://github.com/Azure/azure-sdk-for-python/issues/24652))
```
Comment (mccoyp) [inline sdk/identity/azure-identity/CHANGELOG.md:7]: nit (using the credential name may be useful for searching the changelog):
```suggestion
- EnvironmentCredential added `AZURE_CLIENT_CERTIFICATE_PASSWORD` support for the cert password    ([#24652](https://github.com/Azure/azure-sdk-for-python/issues/24652))
```

**Review by mccoyp — APPROVED** [review]

**Pull request merged**